js_modules/dagit/packages/core/src/partitions/PartitionRangeInput.tsx

```diff
@@ -9,29 +9,40 @@ export const PartitionRangeInput: React.FC<{
   partitionNames: string[];
 }> = ({value, onChange, partitionNames}) => {
   const [valueString, setValueString] = React.useState('');
+  const partitionNameJSON = React.useMemo(() => JSON.stringify(partitionNames), [partitionNames]);
 
   React.useEffect(() => {
-    setValueString(partitionsToText(value, partitionNames));
-  }, [value, partitionNames]);
+    const partitionNameArr = JSON.parse(partitionNameJSON);
```

Comment:
Ahh good catch I think this is a solid fix 👍  I think in the new backfills modal I built for assets the partition names are passed in already memoized, but it's probably not that way everywhere...